app/controllers/api/connections_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::ConnectionsController < ApplicationController
+    include AuthenticationConcern
+
+    def request_match
+
+        if @current_user
+
+            request = MatchRequest.new
+            request.student_id = @current_user.id
+
+            if SkillClass.find_by(id: params[:id])
+                #check if class exists
+                request.class_id = params[:id]
+
+                if MatchRequest.find_by(student_id: @current_user.id, class_id: params[:id])
+                    #check if request already exists
+                    error = {"error": "Match request already exists"}
+                    render(json: error, status: 400)
+                    return
+                end
+
+                if request.save
+                    #success
+
+                    #create respective notification for the teacher
+                    data = {match_id: request.id, person_id: request.skill_class.teacher.id, notification_type: "received_request", 
+                        text: "New match request from #{@current_user.username} to class #{request.skill_class.title}"}
+                    notification = Notification.new(data)
+
+                    if not notification.save
```

Comment:
Instead of `if not`, use `unless`

Reply:
Alternatively, you can do `if notification.save` (where you'd do the render from lines 36-37) and then an `else` block (with the contents of lines 31-33)